examples/quickstart/main.rs

```diff
@@ -114,14 +125,52 @@ async fn main() {
     let mut pairs: HashMap<String, ProtocolComponent> = HashMap::new();
     let mut amounts_out: HashMap<String, BigUint> = HashMap::new();
 
-    let mut protocol_stream = ProtocolStreamBuilder::new(&tycho_url, chain)
-        .exchange::<UniswapV2State>("uniswap_v2", tvl_filter.clone(), None)
-        .exchange::<UniswapV3State>("uniswap_v3", tvl_filter.clone(), None)
-        .exchange::<UniswapV4State>(
-            "uniswap_v4",
-            tvl_filter.clone(),
-            Some(uniswap_v4_pool_with_hook_filter),
-        )
+    let mut protocol_stream = ProtocolStreamBuilder::new(&tycho_url, chain);
+
+    match chain {
+        Chain::Ethereum => {
```

Comment:
maybe we could make this general too and use it in both examples?? or do you think it's an overkill?

Reply:
you mean just this match statement? I think it's not the best idea since execution may at times support fewer protocols than simulation

Reply:
I mean in the quickstart and the price printer!